tests/unit/apps/test_auxiliary.py

```diff
@@ -109,14 +110,21 @@ def test_auxiliary_upgrade_plan_ussuri_to_victoria_change_channel(
     expected_plan = ApplicationUpgradePlan(
         description=f"Upgrade plan for '{app.name}' to {target}"
     )
+    expected_upgrade_package_step = PreUpgradeStep(
+        description=f"Upgrade software packages of '{app.name}' from the current APT repositories",
+        parallel=False,
+    )
+    for unit in app.status.units.keys():
+        expected_upgrade_package_step.add_step(
+            UnitUpgradeStep(
+                description=f"Upgrade software packages on unit {unit}",
+                parallel=False,
+                coro=app_utils.upgrade_packages(unit, model, None),
+            )
+        )
+
     upgrade_steps = [
-        PreUpgradeStep(
-            description=(
-                f"Upgrade software packages of '{app.name}' from the current APT repositories"
-            ),
-            parallel=False,
-            coro=app_utils.upgrade_packages(app.status.units.keys(), model, None),
-        ),
+        expected_upgrade_package_step,
```

Comment:
These tests is getting more and more complicated because of the multi-level steps structure. We should consider refactoring it

Reply:
Totally agree, these tests are not even useful for actual testing.